libs/agno/agno/skills/loaders/db.py

```diff
@@ -0,0 +1,282 @@
+"""Database-backed skill loader.
+
+This loader reads skills from a database and can optionally sync skills back to the database.
+"""
+
+from pathlib import Path
+from typing import List, Optional, Union
+
+from agno.db.base import AsyncBaseDb, BaseDb
+from agno.skills.loaders.base import SkillLoader
+from agno.skills.loaders.local import LocalSkills
+from agno.skills.skill import Skill
+from agno.utils.log import log_debug, log_error, log_info
+
+
+class DbSkills(SkillLoader):
```

Comment:
qq: Do we not want to have the skills path regardless of the type of skill loader? for the first time what's the flow for upserting the skills in the DB if we dont have a way to read them from a file? i'm probably missing something here...

but in my mind the flow is like -> skills directory -> if DB does not have skills present read from that path and upsert -> if present in DB read from DB... (we might have to consider versions here too) 

Does that make sense?

Reply:
Loding skills from DB and from local dir are two completely diff things.  The Local Skills do not sync in DB.
When loading the Skills from DB. The skills are created via API/AgentOS UI
When loading skills from a local dir, skills are just present in local they do not get synced with DB.